server/logs/task-logger.go

```diff
@@ -59,13 +60,18 @@ func StreamTaskLogs(ctx context.Context, streamCfg StreamConfig, restCfg *rest.C
 		return err
 	}
 
+	err = common.WaitForPodToStart(ctx, pods, taskPod)
```

Comment:
I think it makes sense to explain why it is important that we wait for the pod to start before entering the loop.